Navigation menu: Fix alignment, background color, and font size.

ryelle opened the pull request:
See https://github.com/WordPress/wporg-mu-plugins/pull/30#issuecomment-989728416, this should address all the listed issues. It also updates the font size on the large-screen header to match the design (well, it uses `--wp--preset--font-size--small`, which is 14px, not 13px — can use 13px directly if it's important).

> The X button should be centered to its container, it's slightly aligned to the left here

This is to account for the scrollbar, the whole nav is shifted slightly. I don't want to change this, since I don't fully understand why it was done this way.

![small-closed](https://user-images.githubusercontent.com/541093/145456489-fff74ebb-0b92-45f6-8d63-5962d3b4de58.png)
![small-menu](https://user-images.githubusercontent.com/541093/145456490-1881175f-0a27-402b-906c-3528a6a43c00.png)
![small-search](https://user-images.githubusercontent.com/541093/145456491-a8a1e47b-4a1c-4207-8d97-839e853131cf.png)

![large-closed](https://user-images.githubusercontent.com/541093/145456487-391354c1-5387-471e-aa7b-b57d2c529c8f.png)
![large-search](https://user-images.githubusercontent.com/541093/145456488-fad02d1c-11fa-4d41-a4ab-7afabdf4cb23.png)

![Screen Shot 2021-12-09 at 13 46 25](https://user-images.githubusercontent.com/541093/145457030-1a02d4a9-4377-4b63-98e8-3c3428682219.png)

ryelle committed:
Navigation menu: Fix alignment, background color, and font size.

ryelle committed:
Update search font size, fix dropdown background colors

ryelle committed:
Fix menu alignment on desktop size screens

ryelle committed:
Search container should be darker grey on large screens too

ryelle committed:
Remove duplicate border

beafialho commented:
I may not have expressed myself correctly but the menu links on mobile should be `21px`. On desktop they should be `13px`.

@ryelle if you notice here, the dark grey is not quite the same in the icon and screen backgrounds. The correct color value is `#1C2024`

![145456490-1881175f-0a27-402b-906c-3528a6a43c00](https://user-images.githubusercontent.com/65220155/145458053-ba4a6776-1eba-42ec-a472-92e93b0f90d9.png)
![145456491-a8a1e47b-4a1c-4207-8d97-839e853131cf](https://user-images.githubusercontent.com/65220155/145458054-452905ac-0265-4047-873f-9c249ba4d3ca.png)

ryelle committed:
Fix background color of icon button

ryelle commented:
> I may not have expressed myself correctly but the menu links on mobile should be 21px. On desktop they should be 13px.

No, I understood — but I'm using the existing custom properties, so it's 20px on mobile, and 14px on desktop. Are you asking me to change it to 21/13?

Fixed the icon background color (next time could you circle the issue in your screenshot? I usually keep my monitor brightness down, and really can't differentiate #1c2024 from black).

beafialho commented:
> No, I understood — but I'm using the existing custom properties, so it's 20px on mobile, and 14px on desktop. Are you asking me to change it to 21/13?

Yes, please.

> Fixed the icon background color (next time could you circle the issue in your screenshot? I usually keep my monitor brightness down, and really can't differentiate #1c2024 from black).

Sure, I can do that 👍

ryelle committed:
Set font size explicitly to 21px/13px

ryelle committed:
Fix responsive menu script

- Last item in the list should not account for the … menu when looking at available space
- Run the resizing when the screen comes out of mobile/isResponsive mode to avoid glitches
- Assorted code-style cleanup

ryelle commented:
Fixed the font sizes on the menus.

![small](https://user-images.githubusercontent.com/541093/145607460-facc5fda-ef18-4d36-a4c9-35677f028961.png)
![Screen Shot 2021-12-10 at 11 28 16](https://user-images.githubusercontent.com/541093/145607769-6127e309-4bc2-4a79-b99c-05e3321af475.png)

I also fixed what I think was the issue mentioned in https://github.com/WordPress/wporg-news-2021/issues/120, 

> The rule should be: if there is enough space for all the menu links, it shouldn’t collapse into the ... Seems like the margin is too big on the right, it can be just 25px like in its left margin

The last item would still account for the space needed for the … menu item, and collapse if it's not there. But if the last item can fit, we won't need the … item. So now all the menu items can fit at 1280px wide, and it collapses down from there.

![Screen Shot 2021-12-10 at 11 28 48](https://user-images.githubusercontent.com/541093/145607899-3bece3cd-b814-43f6-ba4d-d50a13de222c.png)

![Screen Shot 2021-12-10 at 11 29 18](https://user-images.githubusercontent.com/541093/145607904-ea1d40eb-8daf-495e-be8d-cf8ec27b34b5.png)

ryelle committed:
Navigation script: Only tally the width of the top-level menu items

ryelle commented:
Actually, I had the script a little bit wrong still — it was also tallying the width of the submenu items, which added an extra gap at the end of the menu. Now the menu fills the whole space available, and only starts hiding elements at 1243px wide:

![Screen Shot 2021-12-10 at 11 38 37](https://user-images.githubusercontent.com/541093/145609378-2a1cd422-f60a-416d-8217-e6f51c60eff7.png)

iandunn commented:
> This is to account for the scrollbar, the whole nav is shifted slightly. I don't want to change this, since I don't fully understand why it was done this way.

Yeah, it was because the scrollbar covers part of that space (at least in some browser/OS combinations), making it _look_ like it's not centered.

beafialho commented:
Thank you @ryelle! 

My only comment based on the screenshots: the font sizes for menu items on mobile are correct `21px`, but the submenus should be `15px` as specified in the mockups. This font size change should only affect the menu on mobile, not the desktop menu.

<img width="344" alt="Captura de ecrã 2021-12-15, às 15 34 42" src="https://user-images.githubusercontent.com/65220155/146216749-9b54131a-af2e-4927-a7c9-6d413ae42d1c.png">

ryelle commented:
Updated the font size to 15px in https://github.com/WordPress/wporg-mu-plugins/commit/ea333c269b36494ec9f7e92b3598faccddf244e1:

<img src="https://user-images.githubusercontent.com/541093/146576068-66b50e59-b44a-41de-ad9d-ccd1500eec08.png" width="450" />

beafialho commented:
LGTM 👍
